tests/test_defects.py

```diff
@@ -0,0 +1,270 @@
+
+        flatWidth = np.sqrt(self.flatMean) + self.readNoiseAdu
+        darkWidth = self.readNoiseAdu
+        self.rng = np.random.RandomState(0)
+        flatData = self.rng.normal(self.flatMean, flatWidth, (shapeX, shapeY))
+        darkData = self.rng.normal(self.darkMean, darkWidth, (shapeX, shapeY))
+
+        # NB darks and flats have same defects applied deliberately to both
+        for defect in self.brightDefects:
+            y, x, s = defect
+            flatData[x:x+s, y:y+s] += self.nSigmaBright * flatWidth  # are these actually the numbers we want?
+            darkData[x:x+s, y:y+s] += self.nSigmaBright * darkWidth  # are these actually the numbers we want?
+
+        for defect in self.darkDefects:
+            y, x, s = defect
+            flatData[x:x+s, y:y+s] -= self.nSigmaDark * flatWidth  # are these actually the numbers we want?
+            darkData[x:x+s, y:y+s] -= self.nSigmaDark * darkWidth  # are these actually the numbers we want?
+
+        self.darkExp = self.flatExp.clone()
+        self.spareImage = self.flatExp.clone()  # for testing edge bits and misc
+
+        self.flatExp.image.array[:] = flatData
+        self.darkExp.image.array[:] = darkData
+
+        self.defaultTask = cpPipe.defects.FindDefectsTask(config=self.defaultConfig)
+
+    def tearDown(self):
+        del self
+
+    def test_defectFindingAllSensor(self):
+        config = copy.copy(self.defaultConfig)
+        config.nPixBorderUpDown = 0
+        config.nPixBorderLeftRight = 0
+        task = cpPipe.defects.FindDefectsTask(config=config)
+
+        defects = task.findHotAndColdPixels(self.flatExp, 'flat')
+
+        allBBoxes = self.darkBBoxes + self.brightBBoxes
+
+        for defect in defects:
+            self.assertIn(defect.getBBox(), allBBoxes)
+
+    def test_defectFindingEdgeIgnore(self):
+        task = cpPipe.defects.FindDefectsTask(config=self.defaultConfig)
+        defects = task.findHotAndColdPixels(self.flatExp, 'flat')
+
+        shouldBeFound = self.darkBBoxes[self.noEdges] + self.brightBBoxes[self.noEdges]
+        for defect in defects:
+            self.assertIn(defect.getBBox(), shouldBeFound)
+
+        shouldBeMissed = self.darkBBoxes[self.onlyEdges] + self.brightBBoxes[self.onlyEdges]
+        for defect in defects:
+            self.assertNotIn(defect.getBBox(), shouldBeMissed)
+
+    def test_postProcessDefectSets(self):
+        defects = self.defaultTask.findHotAndColdPixels(self.flatExp, 'flat')
+
+        # minimal test
+        merged = self.defaultTask._postProcessDefectSets([defects], self.flatExp.getDimensions(), 'FRACTION')
+        self.assertEqual(defects, merged)
+
+        # should always be true regardless of config
+        merged = self.defaultTask._postProcessDefectSets([defects, defects], self.flatExp.getDimensions(),
+                                                         'FRACTION')
+        self.assertEqual(defects, merged)
```

Comment:
How is this different from the above test?

Reply:
One is "merging" a set with nothing (the list has length one), the second is merging it with itself. This was failing at one point, so I know it's not identical!

Reply:
I've now added a comment to point that out though 👍